core/common/scale.hpp

```diff
@@ -18,21 +18,33 @@
  * When decoding you have to reinterpret integer value as signed
  * if you know it is signed.
  *
- * This library provides following options:
- * 1. encode/decode fixed width integers: 1,2,4 byte integers:
- *    it only translates them from host to little-endian and vice-versa
- * 2. encode/decode boolean values: true is encoded as 0x01,false as 0x00
- * 3. encode/decode tristate values: none is 0x00, false is 0x01, true is 0x02
- * 4. encode/decode one-byte options
- * 5. collections of same type values are encoded/ decoded by following
- * scheme: first value is compact-encoded number of items followed by each
- * item scale-encoded sequencially
- * 6. byte arrays are example of collections
- * 7. enumerations
+ * This library provides encoding and decoding following types:
+ * 1. Fixed width signed and unsigned 1,2,4 byte integers:
+ * 2. Unsigned big integers in range 2^30..2^536-1
+ * 2. Boolean values: true is encoded as 0x01,false as 0x00
+ * 3. Tribool values: false is 0x00, true is 0x01, indeterminate is 0x02
```

Comment:
Not sure if we should think like this. 
By options they mean optional values and encode them as follows:
first byte can be 0 or 1. When 0 it means, that no value exists, 1 means that it does. After the flag byte encoded value itself is placed. And here they consider exceptional case of encoding optional bool value by one byte instead of 2. Not sure if optional bool is equal to tribool. I used to think that  optionals are used to show that an error occured and no value present, but tribool is basis of complex logic. I didn't implement this "exceptional" case and don't know should I?

Reply:
Added specialization for encode|decodeOptional for bool type, and tests as well.